Plugins-MQ/src/main/java/com/greicodex/openas2/plugins/mq/MessageBuilderModule.java

```diff
@@ -0,0 +1,167 @@
+package com.greicodex.openas2.plugins.mq;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.activation.DataHandler;
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeBodyPart;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openas2.OpenAS2Exception;
+import org.openas2.Session;
+import org.openas2.WrappedException;
+import org.openas2.message.FileAttribute;
+import org.openas2.message.Message;
+import org.openas2.params.CompositeParameters;
+import org.openas2.params.DateParameters;
+import org.openas2.params.InvalidParameterException;
+import org.openas2.params.MessageParameters;
+import org.openas2.params.ParameterParser;
+import org.openas2.params.RandomParameters;
+import org.openas2.partner.Partnership;
+import org.openas2.processor.receiver.BaseReceiverModule;
+import org.openas2.util.AS2Util;
+
+public abstract class MessageBuilderModule extends BaseReceiverModule {
+
```

Comment:
Why are you re-implementing the MessageBuilderModule?
If there is something that you need in the existing one then lets talk about enhancing it rather than duplicating all the code.